### PR TITLE
Fix mistakes and style in docs

### PR DESCRIPTION
**Pull request by irinakhismatullina**
Based on #625. Forth and final part of #615.

### CODE REVIEW

File: lookout/style/typos/analyzer.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 """Identifier typos analyzer."""
+
```
Comment (zurk) [inline lookout/style/typos/analyzer.py]: we do not put a newline after module docstring.

File: lookout/style/typos/analyzer.py [hunk 2]
```diff
@@ -153,9 +154,9 @@ def filter_suggestions(self, test_df: pandas.DataFrame,
         Filter suggestions based on the repo specifics and confidence threshold.
 
         :param test_df: DataFrame with info about tested tokens.
-        :param suggestions: Dictionary of correction suggestions grouped by \
+        :param suggestions: Dictionary of correction suggestions grouped by
```
Comment (zurk) [inline lookout/style/typos/analyzer.py]: I think you should use ` \` because otherwise, param docsting will end here. But I am not 100% sure. Can you check it or read in doc?

File: lookout/style/typos/generation.py [hunk 3]
```diff
@@ -114,8 +116,9 @@ def generate_candidates(self, data: pandas.DataFrame, threads_number: int,
         else:
             candidates = [self._lookup_corrections_for_token(t) for t in typos]
         candidates = pandas.DataFrame(list(chain.from_iterable(candidates)))
-        candidates.columns = [Columns.Id, Columns.Token, Columns.Candidate, Columns.Features]
-        candidates[Columns.Id] = candidates[Columns.Id].astype(data.index.dtype)
+        candidates.columns = [Columns.Id, Columns.Token, Columns.Candidate,
+                              Columns.Features]
```
Comment (zurk) [inline lookout/style/typos/generation.py]: fits into one line, why did you change it?

File: lookout/style/typos/utils.py [hunk 4]
```diff
@@ -20,42 +21,50 @@ def read_frequencies(file: str) -> Dict[str, int]:
...
-            split = line.split()
-            frequencies[split[0]] = int(split[1])
+    try:
+        f = lzma.open(file, "rt")
+    except lzma.LZMAError:
+        f = open(file, "r")
+    for line in f:
+        split = line.split()
+        frequencies[split[0]] = int(split[1])
+    f.close()
```
Comment (zurk) [inline lookout/style/typos/utils.py]: you should always close the file even if exception raises after you open the file. Please use the file extension to check file format 
```python
    open_ = lzma.open if file.endswith(".xz") else open
    with open_(file, "rt"):
        ...
```

and why `"rt"` and not `"rb"`?

File: lookout/style/typos/utils.py [hunk 5]
```diff
@@ -20,42 +21,50 @@ def read_frequencies(file: str) -> Dict[str, int]:
...
     :return: List of tokens of the vocabulary.
     """
-    with lzma.open(file, "rt") as f:
-        tokens = [line.split()[0] for line in f]
+    try:
+        f = lzma.open(file, "rt")
+    except lzma.LZMAError:
+        f = open(file, "r")
+    tokens = [line.split()[0] for line in f]
+    f.close()
```
Comment (zurk) [inline lookout/style/typos/utils.py]: same here.

File: lookout/style/typos/utils.py [hunk 6]
```diff
@@ -20,42 +21,50 @@ def read_frequencies(file: str) -> Dict[str, int]:
...
     return tokens
 
 
 def flatten_df_by_column(data: pandas.DataFrame, column: str, new_column: str,
                          apply_function=lambda x: x) -> pandas.DataFrame:
     """
-    Flatten DataFrame by `column` with extracted elements put to `new_column`. \
-    Operation happens out-of-place.
+    Flatten DataFrame by `column` with extracted elements put to `new_column` (out-of-place).
```
Comment (zurk) [inline lookout/style/typos/utils.py]: I was better on my taste.

File: lookout/style/typos/analyzer.py (hunk 1 above)
Comment (irinakhismatullina) [reply lookout/style/typos/analyzer.py]: Thank you, fixed it.

File: lookout/style/typos/analyzer.py (hunk 2 above)
Comment (irinakhismatullina) [reply lookout/style/typos/analyzer.py]: Yes, it's a mistake.

File: lookout/style/typos/utils.py (hunk 4 above)
Comment (irinakhismatullina) [reply lookout/style/typos/utils.py]: This `lzma` thing will be removed complitely in future PRs, can we just skip it here?

Comment (irinakhismatullina) [reply lookout/style/typos/utils.py]: Both `'r'` and `'rb'` work there actually, I think `'rt'` just stuck somehow.

Comment (irinakhismatullina) [reply lookout/style/typos/utils.py]: So should I change these chunks, considering that they will be removed in the future?

File: lookout/style/typos/utils.py [hunk 7]
```diff
@@ -32,42 +32,51 @@ def read_frequencies(file: str) -> Dict[str, int]:
...
+    :param file: Path to the file containing space-separated word-frequency pairs one-per-line. \
+                 File should be in .csv or compressed .csv.xz format.
     :return: Dictionary of tokens frequencies.
     """
     frequencies = {}
-    with lzma.open(file, "rt") as f:
-        for line in f:
-            split = line.split()
-            frequencies[split[0]] = int(split[1])
+    try:
```
Comment (vmarkovtsev) [inline lookout/style/typos/utils.py]: Let's use https://github.com/RaRe-Technologies/smart_open

Comment (vmarkovtsev) [reply lookout/style/typos/utils.py]: I will submit xz support there, let's use bzip2 for now

File: lookout/style/typos/utils.py [hunk 8]
```diff
@@ -32,42 +32,51 @@ def read_frequencies(file: str) -> Dict[str, int]:
...
+    f.close()
     return tokens
 
 
 def flatten_df_by_column(data: pandas.DataFrame, column: str, new_column: str,
                          apply_function=lambda x: x) -> pandas.DataFrame:
     """
     Flatten DataFrame by `column` with extracted elements put to `new_column`. \
-    Operation happens out-of-place.
+    Operation happels out-of-place.
```
Comment (vmarkovtsev) [inline lookout/style/typos/utils.py]:
```suggestion
    Operation happens out-of-place.
```

File: lookout/style/typos/utils.py (hunk 7 above)
Comment (irinakhismatullina) [reply lookout/style/typos/utils.py]: This .xz support will be removed completely in the upcoming PRs. I still should change this?

Comment (vmarkovtsev) [reply lookout/style/typos/utils.py]: Then cool, use this package as-is.

FWIW https://github.com/RaRe-Technologies/smart_open/pull/260

Comment (irinakhismatullina) [reply lookout/style/typos/utils.py]: As-is it just doesn't work. Or I don't understand what you mean by "as-is", or I will have to change formats of several files from tests, and then change them back in the upcoming PRs.

**Comment by vmarkovtsev** [comment]
@irinakhismatullina Conflict here

**Comment by irinakhismatullina** [comment]
@vmarkovtsev Conflict resolved, only tests with smart_open don't pass as expected.

**Comment by vmarkovtsev** [comment]
Because of xz, right?

**Comment by irinakhismatullina** [comment]
Yes

**Comment by vmarkovtsev** [comment]
It will take some for the PR to be merged and the new release published. Can you please add this to `requirementx.txt`:

```
-e git://github.com/vmarkovtsev/smart_open#egg=smart_open
```

and leave `smart_open` in setup.py.

**Comment by vmarkovtsev** [comment]
Yes! Finally.
